e2e/utils/ProfileUtils.ts

```diff
@@ -0,0 +1,370 @@
+    }
+  } catch {
+    return [];
+  }
+}
+
+export async function clearUserProfile(): Promise<string[]> {
+  const platform = os.platform() as 'win32' | 'darwin' | 'linux';
+  const skipReasons: string[] = [];
+
+  if (platform === 'win32') {
+    return await verifyNoRegistryHive('HKCU');
+  }
+  const profilePaths = getDeploymentPaths(platform, paths.deploymentProfileUser);
+
+  for (const fullPath of profilePaths) {
+    try {
+      await fs.promises.rm(fullPath, { force: true });
+    } catch (ex: any) {
+      skipReasons.push(`Failed to delete file ${ fullPath }: ${ ex }`);
+    }
+  }
+
+  return skipReasons;
+}
+
+async function fileExists(fullPath: string): Promise<boolean> {
+  try {
+    await fs.promises.access(fullPath);
+
+    return true;
+  } catch { }
+
+  return false;
+}
+
+function getDeploymentBaseNames(platform: 'linux'|'darwin'): string[] {
+  if (platform === 'linux') {
+    return ['rancher-desktop.defaults.json', 'rancher-desktop.locked.json'];
+  } else if (platform === 'darwin') {
+    return ['io.rancherdesktop.profile.defaults.plist', 'io.rancherdesktop.profile.locked.plist'];
+  } else {
+    throw new Error(`Unexpected platform ${ platform }`);
+  }
+}
+
+function getDeploymentPaths(platform: 'linux'|'darwin', profileDir: string): string[] {
+  return getDeploymentBaseNames(platform).map(basename => path.join(profileDir, basename));
+}
+
+export async function hasRegistryHive(hive: string): Promise<boolean> {
+  for (const profileType of ['defaults', 'locked']) {
+    try {
+      const { stdout } = await childProcess.spawnFile('reg',
+        ['query', `${ hive }\\SOFTWARE\\Policies\\Rancher Desktop\\${ profileType }`],
+        { stdio: ['ignore', 'pipe', 'pipe'] });
+
+      if (stdout.length > 0) {
+        return true;
+      }
+    } catch { }
+  }
+
+  return false;
+}
+
+export async function hasUserProfile(): Promise<boolean> {
+  const platform = os.platform() as 'win32' | 'darwin' | 'linux';
+
+  if (platform === 'win32') {
+    return await hasRegistryHive('HKCU');
+  }
+
+  for (const profilePath of getDeploymentPaths(platform, paths.deploymentProfileUser)) {
+    try {
+      await fs.promises.access(profilePath);
+
+      return true;
+    } catch { }
+  }
+
+  return false;
+}
+
+export async function verifyRegistryHive(hive: string): Promise<string[]> {
+  let hasProfile = false;
+
+  for (const profileType of ['defaults', 'locked']) {
+    try {
+      const { stdout } = await childProcess.spawnFile('reg',
+        ['query', `${ hive }\\SOFTWARE\\Policies\\Rancher Desktop\\${ profileType }`],
+        { stdio: ['ignore', 'pipe', 'pipe'] });
+
+      if (stdout.length > 0) {
+        hasProfile = true;
+        break;
+      }
+    } catch { }
+  }
+
+  return hasProfile ? [] : [`Need to add registry hive "${ hive }\\SOFTWARE\\Policies\\Rancher Desktop\\<defaults or locked>"`];
+}
+
+export async function verifySettings(): Promise<string[]> {
```

Comment:
I'd prefer to stick with `verify` because when we run into something that we can't verify, we either throw an exception or return a reason to skip the current test.